### DISCUSSION TITLE
inductance and resistance

### DISCUSSION THREAD

**Discussion by faraz4714** [Eomys/pyleecan#622] [General]
Hello Pierre 
Thank you for your help in advance 
I am simulating an IPMSM motor and I want to calculate inductance and resistance (Ld, Lq, R1) and I saw a class EEC_PMSM that Tsat = 20 and Trot = 20 but I want to be sure those values are calculated at 20 deg or different temperature and also R1 it seems equivalent phase resistance am I right?

Best regards,
Faraz Mirali

**Comment by BonneelP** [comment]
Hello,

Tsat and Trot are input that you can set/change (there is also T_mag for FEMM). Regarding the electrical module, you can take a look to pyleecan\Methods\Simulation\EEC_PMSM\comp_parameters.py and pyleecan\Methods\Simulation\Electrical\run.py to understand how the module works. We also have a tutorial but we still need to understand why the last plot is wrong: https://pyleecan.org/05_tuto_Elec.html
According to pyleecan\Methods\Simulation\EEC\comp_R1.py, R1 is "stator phase resistance for the equivalent electrical circuit"

Best regards,
Pierre

**Comment by faraz4714** [comment]
hello,
I can't understand some things and I don't know how to solve them. when I change I0_rms and Phi_0 and also T_mag in this code, the output for Ld and Lq didn't change (I want to calculate Ld and Lq at 20 deg and because of that I didn't change the default value for Tsat and Trot) however these value is far from different from real value that I measured in real case

```
from pyleecan.Classes.Simu1 import Simu1
from pyleecan.Classes.Electrical import Electrical
from pyleecan.Classes.EEC_PMSM import EEC_PMSM
from pyleecan.Classes.MagFEMM import MagFEMM

# Initialization of the Simulation
simu = Simu1(name="tuto_elec", machine=IPMSM_A)
# Definition of the Electrical Equivalent Circuit
eec = EEC_PMSM(  # No parameter enforced => compute all
    fluxlink=MagFEMM(is_periodicity_a=True, T_mag = 120),  # Magnetic model to compute fluxlinkage
)
# The Electrical module is defined with the EEC
simu.elec = Electrical(eec=eec)

# Run only Electrical module
simu.mag = None
simu.force = None
simu.struct = None

from pyleecan.Classes.InputCurrent import InputCurrent
from pyleecan.Classes.OPdq import OPdq
from numpy import sqrt, exp, pi

# Definition of a sinusoidal current
simu.input = InputCurrent()

I0_rms = 14.019/sqrt(3) # Maximum current [Arms]
Phi0 = 96.1666666 *pi/180  # Maximum Torque Per Amp

simu.input.OP = OPdq(N0 = 4300)
simu.input.OP.set_I0_Phi0(I0_rms, Phi0)

simu.input.Nt_tot = 150 # Number of time steps
simu.input.Na_tot = 2048 # Spatial discretization

(simu.input.OP.Id_ref,simu.input.OP.Iq_ref)
```

Best regards,
Faraz Mirali

**Reply by BonneelP** [reply]
Hello,

Your definition looks right to me. Maybe the simulation is modified during the call to simu.run and then running the simulation twice don't recompute everything. You can try the following:
```python
simu2 = simu.copy()
out = simu.run()
simu2.compare(simu)
```

You can also check the log of both simulation to see if the messages are different (for instance are the EEC parameters computed in both simulation). Anyway we know that the electrical module needs some rework, in particular to understand the error on the tutorial last plot (https://pyleecan.org/05_tuto_Elec.html) so maybe the best is to run the code step by step with breakpoints and check what is going on. Sadly, I won't have the time to do that in the near future.

Best regards,
Pierre

**Comment by faraz4714** [comment]
Hello,
Thank you for your answer. I solve it.
but I have a question in theory. I saw this class "pyleecan/Methods/Simulation/EEC_PMSM/comp_Phidq.py" and I found this equation 
Phiq = self.Lq * Idq_dict["Iq"] + self.Phiq_mag
I want to know why for calculating Phiq you add flux linkage for the permanent magnet because the D axis is directly along the permanent magnet flux so the Q axis is 90 degrees out of phase from this flux 
can you explain me, please? 
best regards,
Faraz Mirali

**Reply by BonneelP** [reply]
Hello,

You may have found the typo in the Electrical module, I think that the correct equation should be: Phiq = LqIq and Phid = LdId + Phi_mag
Can you try to change the formula and check if the results are better ? If yes can you open a PR ? 

Best regards,
Pierre

**Reply by EmileDvs** [reply]
Hello,

The interest of keeping phiq_mag might have been in case of uneven magnetization for exemple. In any case it doesnt make the ideal case wrong since the quantity is zero. 

Beet regards
Emile

**Comment by faraz4714** [comment]
Hello,
thank you for your answer.
I try it somethings, in my case,Phiq_mag it's 3,4 times magnitude less than phiq so it can be negligible. 

Best regards,
Faraz Mirali
